apps/explorer/src/components/owned-objects/views/FieldItem.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiMoveNormalizedType } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import { getFieldTypeValue } from '~/components/owned-objects/utils';
+import { AddressLink, ObjectLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { Text } from '~/ui/Text';
+
+interface FieldItemProps {
+    value: string | number | object | boolean;
+    type: SuiMoveNormalizedType;
+    truncate?: boolean;
+}
+
+const TYPE_ADDRESS = 'Address';
+const TYPE_URL = '0x2::url::Url';
+const TYPE_OBJECT_ID = ['0x2::object::UID', '0x2::object::ID'];
+
+export function FieldItem({ value, type, truncate = false }: FieldItemProps) {
+    // for object types, use SyntaxHighlighter
+    if (typeof value === 'object') {
+        return (
+            <SyntaxHighlighter
+                code={JSON.stringify(value, null, 2)}
+                language="json"
+            />
+        );
+    }
+
+    const { displayName, normalizedType } = getFieldTypeValue(type);
+
+    if (displayName === TYPE_ADDRESS) {
+        return (
+            <div className="break-all">
+                <AddressLink
+                    address={value.toString()}
+                    noTruncate={!truncate}
```

Comment:
Curious in what cases we don't just truncate these? Fine with this just curious

Reply:
Whenever the DisclosureBox is close.  
<img width="1124" alt="Screenshot 2023-04-13 at 3 11 46 PM" src="https://user-images.githubusercontent.com/126525197/231859804-f21e8fae-e05f-4090-80d3-1f62df1e804c.png">